glean-core/csharp/Glean/Glean.cs

```diff
@@ -183,6 +183,75 @@ internal bool IsInitialized()
             return initialized;
         }
 
+        /// <summary>
+        /// Enable or disable Glean collection and upload.
+        /// 
+        /// Metric collection is enabled by default.
+        /// 
+        /// When uploading is disabled, metrics aren't recorded at all and no data
+        /// is uploaded.
+        /// 
+        /// When disabling, all pending metrics, events and queued pings are cleared.
+        /// 
+        /// When enabling, the core Glean metrics are recreated.
+        /// </summary>
+        /// <param name="enabled">When `true`, enable metric collection.</param>
+        public void SetUploadEnabled(bool enabled)
+        {
+            if (IsInitialized())
+            {
+                bool originalEnabled = GetUploadEnabled();
+    
+                Dispatchers.LaunchAPI(() => {
+                LibGleanFFI.glean_set_upload_enabled(enabled);
+
+                if (!enabled)
+                {
+                    // Cancel any pending workers here so that we don't accidentally upload or
```

Comment:
This should be able to use the `Dispatchers.CancelBackgroundTasks()` once I'm done with it.

Reply:
I think that will depend on how the uploader would look like :)